dockerfile_build/config_stripper.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/python
+
+# Copyright 2017 Google Inc. All rights reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import argparse
+import json
+import os
+import shutil
+import sys
+import tarfile
+import tempfile
+
+_TIMESTAMP = '1970-01-01T00:00:00Z'
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--in_tar_path', type=str,
+                        help='Path to docker save tarball',
+                        required=True)
+    parser.add_argument('--out_tar_path', type=str,
+                        help='Path to output stripped tarball',
+                        required=True)
+    args = parser.parse_args()
+
+    return strip_tar(args.in_tar_path, args.out_tar_path)
+
+
+def strip_tar(input, output):
+    # Unpack the tarball, modify configs in place, and rearchive.
+    # We need to take care to keep the files sorted.
+
+    tempdir = tempfile.mkdtemp()
+    with tarfile.open(name=input, mode='r') as it:
+        it.extractall(tempdir)
+    
+    with open(os.path.join(tempdir, 'manifest.json'), 'r') as mf:
```

Comment:
Can we make this go through `google/containerregistry` abstractions instead of teaching another place the `docker save` tarball format?

Reply:
Just poked around this a bit: the containerregistry abstractions aren't designed to be mutable  (I'd need to poke at _config_file of docker_image.FromTarball).

Using Overrides with append won't work, because that's just appending a repeatable config on top of a non-repeatable one.

Any suggestions? Maybe subclassing FromTarball and "monkey-patching" the config_blob/config_file methods to strip timestamps?